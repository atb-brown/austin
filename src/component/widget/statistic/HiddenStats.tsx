import "../../../common.css";
import "./Stats.css";
import React from "react";

/**
 * A component that is the "hidden" banner, which is basically a button that
 * would, in context, expand back to the full stats ribbon.
 *
 * @return {ReactElement} React Component
 */
export default function Stats({ onClick }: HiddenStatsProps) {
  const leftArrow = "â—€";
  return (
    <div className="Stats Stats-Container disable-text-selection">
      <div
        className="Stats Stats-Visual"
        data-testid="stats-ribbon-hidden"
        onClick={onClick}
        style={{
          aspectRatio: 1,
          justifyContent: "center",
        }}
      >
        {leftArrow}
      </div>
    </div>
  );
}

/**
 * Properties for the HiddenStats component.
 */
type HiddenStatsProps = {
  readonly onClick: () => void;
};
