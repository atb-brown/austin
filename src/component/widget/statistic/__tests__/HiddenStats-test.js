import HiddenStats from "../HiddenStats";
import "@testing-library/jest-dom";
import { screen, render, fireEvent } from "@testing-library/react";

it("initial view renders to match snapshot", () => {
  const { asFragment } = render(<HiddenStats />);

  expect(asFragment()).toMatchSnapshot();
});

it("onClick is invoked when component is clicked", () => {
  let clickCount = 0;
  const onClick = () => {
    clickCount++;
  };

  const { asFragment } = render(<HiddenStats onClick={onClick} />);

  expect(asFragment()).toMatchSnapshot();
  fireEvent.click(screen.getByTestId("stats-ribbon-hidden"));
  expect(clickCount).toBe(1);
});
