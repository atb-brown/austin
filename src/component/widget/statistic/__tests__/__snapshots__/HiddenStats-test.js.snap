// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial view renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Stats Stats-Container disable-text-selection"
  >
    <div
      class="Stats Stats-Visual"
      data-testid="stats-ribbon-hidden"
      style="justify-content: center;"
    >
      ◀
    </div>
  </div>
</DocumentFragment>
`;

exports[`onClick is invoked when component is clicked 1`] = `
<DocumentFragment>
  <div
    class="Stats Stats-Container disable-text-selection"
  >
    <div
      class="Stats Stats-Visual"
      data-testid="stats-ribbon-hidden"
      style="justify-content: center;"
    >
      ◀
    </div>
  </div>
</DocumentFragment>
`;
